src/app/dao/tests/UserDAO.spec.ts

```diff
@@ -79,4 +84,131 @@ describe('UserDAO', () => {
+            const token: string = '{"sub": "foo@bar.com", "email": "foo@bar.com", "email_verified": true}';
+            const credential: FireAuth.UserCredential =
+                await FireAuth.signInWithCredential(TestBed.inject(FireAuth.Auth),
+                                                    FireAuth.GoogleAuthProvider.credential(token));
+            await dao.set(credential.user.uid, { verified: false, username: 'user' });
+
+            // When marking the user as verified
+            const result: Promise<void> = dao.markVerified(credential.user.uid);
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid setting the user to verified if it has no username', async() => {
+            // Given a non-verified user, without a username
+            const token: string = '{"sub": "foo@bar.com", "email": "foo@bar.com", "email_verified": true}';
+            const credential: FireAuth.UserCredential =
+                await FireAuth.signInWithCredential(TestBed.inject(FireAuth.Auth),
+                                                    FireAuth.GoogleAuthProvider.credential(token));
+            await dao.set(credential.user.uid, { verified: false });
+
+            // When marking the user as verified
+            const result: Promise<void> = dao.markVerified(credential.user.uid);
+            // Then it should fail
+            await expectFirebasePermissionDenied(result);
+        });
+        it('should forbid setting the user to verified if it has no verified email', async() => {
+            // Given a email user that has not verified its email
+            const credential: FireAuth.UserCredential =
+                await FireAuth.createUserWithEmailAndPassword(TestBed.inject(FireAuth.Auth),
+                                                              'foo@bar.com',
+                                                              'jeanjaja123');
+            await dao.set(credential.user.uid, { verified: false, username: 'foo' });
+
+            // When marking the user as verified
+            const result: Promise<void> = dao.markVerified(credential.user.uid);
+            // Then it should fail
+            await expectFirebasePermissionDenied(result);
+        });
+        it('should forbid to update the fields another user', async() => {
+            // Given an existing user and a logged in user
+            const other: FireAuth.User = await createConnectedGoogleUser('bar@bar.com', 'other-user');
+            await signOut();
+            await createConnectedGoogleUser('foo@bar.com', 'user');
+            // When trying to change a field of another user
+            const result: Promise<void> = dao.update(other.uid, { username: 'jean? jaja!' });
+            // Then it should fail
+            await expectFirebasePermissionDenied(result);
+        });
+        it('should forbid to delete its own user', async() => {
```

Comment:
does not sound to RGDB like this :p

Reply:
RGDB :D You mean RGPD mais t'as pas envie de dire les deux dernières lettres ? :D 
Sinon sous RGPD tu dois avoir la possibilité de supprimer tes données sur demande, donc c'pas un soucis ici !

Reply:
Bah là c'est interdit de supprimer tes données :p
Du coup il leur reste que l'email à l'un d'entre nous :p ?

Reply:
oui, donc c'est bon !

Reply:
Mais pour l'instant y a pas de page "contact" sur le site :p !

Reply:
ça sera dans le footer qui est ticketté v1 !